iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -289,23 +382,35 @@ public static Network read(InputStream is) {
         return read(is, new ImportOptions(), null);
     }
 
+    public static void runEndTasks(NetworkXmlReaderContext context, Set<String> extensionNamesNotFound, ImportOptions config) {
+        context.getEndTasks().forEach(Runnable::run);
+
+        if (!extensionNamesNotFound.isEmpty()) {
+            if (config.isThrowExceptionIfExtensionNotFound()) {
+                throw new PowsyblException("Extensions " + extensionNamesNotFound + " not found !");
+            } else {
+                LOGGER.error("Extensions {} not found", extensionNamesNotFound);
+            }
+        }
```

Comment:
You never check this when you're not in a mode different than `NO_SEPARATED_FILE_FOR_EXTENSIONS`